### Importing Needed Packages

In [37]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

### Reading data from csv

In [15]:
df = pd.read_csv('Heart.csv')
df

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0     63    1   3     145   233    1        0       150     0      2.3    0   
1     37    1   2     130   250    0        1       187     0      3.5    0   
2     41    0   1     130   204    0        0       172     0      1.4    2   
3     56    1   1     120   236    0        1       178     0      0.8    2   
4     57    0   0     120   354    0        1       163     1      0.6    2   
..   ...  ...  ..     ...   ...  ...      ...       ...   ...      ...  ...   
298   57    0   0     140   241    0        1       123     1      0.2    1   
299   45    1   3     110   264    0        1       132     0      1.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   
302   57    0   1     130   236    0        0       174     0      0.0    1   

     caa  thall  output  
0      0      1       1  
1      0      2       1  
2      0      2       1  
3      0      2       1  
4      0      2       1  
..   ...    ...     ...  
298    0      3       0  
299    0      3       0  
300    2      3       0  
301    1      3       0  
302    1      2       0  

[303 rows x 14 columns]

### Check for null value

In [17]:
df.isnull().sum()

age         0
sex         0
cp          0
trtbps      0
chol        0
fbs         0
restecg     0
thalachh    0
exng        0
oldpeak     0
slp         0
caa         0
thall       0
output      0
dtype: int64

### Distribution of output column

In [24]:
df['output'].value_counts()

output
1    165
0    138
Name: count, dtype: int64

* we have 165 samples for sick and 138 samples for healthy

### Splitting the Features and Traget

In [29]:
X = df.drop(columns = 'output', axis = 1)
Y = df['output']

### Splitting the Data into Train Data and Test Data

In [31]:
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.2, stratify = Y)

# Model Training

### Logistic Regression

In [38]:
Model = LogisticRegression()

### Training the Logistic Regression Model whit Train Data

In [41]:
Model.fit(X_train, Y_train)

/Users/alireza/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

### Model Evaluation

### Accuracy on train data

In [49]:
X_train_ = Model.predict(X_train)
train_data_accuracy = accuracy_score(X_train_, Y_train)

print("Accuracy on Training Data:", train_data_accuracy)

Accuracy on Training Data: 0.859504132231405


### Accuracy on test data

In [52]:
X_test_ = Model.predict(X_test)
test_data_accuracy = accuracy_score(X_test_, Y_test)

print("Accuracy on Training Data:", test_data_accuracy)

Accuracy on Training Data: 0.819672131147541
